pkg/images/images.go

```diff
@@ -0,0 +1,237 @@
+	return image.ToString()
+}
+
+// GetLatestClusterAgentImage return the latest stable agent release version
+func GetLatestClusterAgentImage() string {
+	image := newImage(DefaultImageRegistry, DefaultClusterAgentImageName, ClusterAgentLatestVersion, false, false)
+	return image.ToString()
+}
+
+// AssembleImage builds the image string based on ImageConfig and the registry configuration.
+func AssembleImage(imageSpec *v2alpha1.AgentImageConfig, registry string) string {
+	if imageNameContainsTag(imageSpec.Name) {
+		return imageSpec.Name
+	}
+
+	if registry == "" {
+		registry = DefaultImageRegistry
+	}
+
+	tag := imageSpec.Tag
+	// If JMXEnabled, then proactively trim JMX suffix to prevent duplicate suffixes
+	if imageSpec.JMXEnabled {
+		tag = strings.TrimSuffix(imageSpec.Tag, JMXTagSuffix)
+	}
+
+	img := newImage(registry, imageSpec.Name, tag, imageSpec.JMXEnabled, false)
+
+	return img.ToString()
+}
+
+// OverrideAgentImage takes an existing image reference and potentially overrides portions of it based on the provided image configuration
+func OverrideAgentImage(currentImage string, overrideImageSpec *v2alpha1.AgentImageConfig) string {
+	image := FromString(currentImage)
+	overrideImage := fromImageConfig(overrideImageSpec)
+
+	image.WithRegistry(overrideImage.registry).
+		WithName(overrideImage.name).
+		WithTag(overrideImage.tag).
+		WithJMX(overrideImage.isJMX).
+		WithFIPS(overrideImage.isFIPS)
+
+	return image.ToString()
+}
+
+// String return the string representation of an image
+func (i *Image) ToString() string {
+	suffix := ""
+	if i.isFIPS {
+		suffix = FIPSTagSuffix
+	}
+	if i.isJMX {
+		suffix = suffix + JMXTagSuffix
+	}
+	return fmt.Sprintf("%s/%s:%s%s", i.registry, i.name, i.tag, suffix)
+}
+
+func FromString(stringImage string) *Image {
+	splitImg := strings.Split(stringImage, "/")
+	registry := strings.Join(splitImg[:len(splitImg)-1], "/")
+
+	splitName := strings.Split(splitImg[len(splitImg)-1], ":")
+
+	name := splitName[0]
+	tag := splitName[1]
+
+	// Check if this tag has JMX or FIPS suffixes
+	// JMX would be on the outside
+	isJMX := strings.HasSuffix(tag, JMXTagSuffix)
+	if isJMX {
+		tag = strings.TrimSuffix(tag, JMXTagSuffix)
+	}
+
+	isFIPS := strings.HasSuffix(tag, FIPSTagSuffix)
+	if isFIPS {
+		tag = strings.TrimSuffix(tag, FIPSTagSuffix)
+	}
+
+	return newImage(registry, name, tag, isJMX, isFIPS)
+}
+
+// fromImageConfig creates an Image instance from the AgentImageConfig spec object
+// We accept the imageConfig.name in the following formats:
+// - name
+// - name:tag
+// - registry/name:tag
+// (Notably, we do not accept "registry/name".)
+// Note that if the name includes a tag, then we ignore imageConfig.tag and imageConfig.JMXEnabled
+func fromImageConfig(imageConfig *v2alpha1.AgentImageConfig) *Image {
+	registry := ""
+	imageName := imageConfig.Name
+	imageTag := imageConfig.Tag
+
+	nameContainsTag := false
+	isJMX := false
+	isFIPS := false
```

Comment:
this will always be false unless `fips` is in tag so only settings `jmxEnabled` in overrdie won't produce `fips-jmx` image, not sure if it should.

Reply:
thanks, yea it's expected that if the tag is included `jmxEnabled` will be ignored (as noted on line 200). however it's a good point that `isFIPS` will always be false then no matter what, so i think i will remove this bool to reduce confusion